src/agreement.rs

```diff
@@ -19,6 +21,116 @@ error_chain!{
     }
 }
 
+/// A lattice-valued description of the state of `bin_values`, essentially the same as the set of
+/// subsets of `bool`.
+#[cfg_attr(feature = "serialization-serde", derive(Serialize))]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum BinValues {
```

Comment:
I tried [bool; 2] at first but replaced it with a specialised enum because you need to define operations anyway, `insert`, `union`, and also `from_iter` for `count_conf`, and those actually take the most of code. Maybe move the enum and the methods to a separate module?

Reply:
Yes, if we need this I'd also put it in a separate module. (Maybe a submodule of `agreement` if it's only used here.)